packages/common/src/hooks/useCoinflowAdapter.ts

```diff
@@ -14,50 +20,110 @@ type CoinflowAdapter = {
   connection: Connection
 }
 
-export const useCoinflowAdapter = () => {
+type UseCoinflowAdapterArgs = {
+  onSendTransaction?: (transaction: Transaction) => Promise<Transaction>
+}
+
+type TransactionDecorator = (transaction: Transaction) => Promise<Transaction>
+
+export const useCoinflowWithdrawalDecorator = () => {
```

Comment:
nit: maybe just have a `useCoinflowWithdrawalAdapter` that returns the full adapter?

Reply:
Could do! But we don't decorate the transactions for purchases. So we have to differentiate somehow?

Reply:
I'm thinking we'd have two separate adapters - `useCoinflowAdapter` and `useCoinflowWithdrawalAdapter`, or if you'd rather `useCoinflowPurchaseAdapter` in place of a root `useCoinflowAdapter`, though I think it's unnecessary

Reply:
Ah okay. That probably makes more sense. I guess I was leaving it open for us to be able to choose at runtime to decorate or not. But I think we're pretty locked in on the flow. I'll split it into two adapters.